Team8923_2020/src/main/java/org/firstinspires/ftc/team8923_2020/MasterOpMode.java

```diff
@@ -3,133 +3,102 @@
-
-        // gets the largest power
-        double scaleFactor = Math.max(Math.max(powerFL, powerFR), Math.max(powerBL, powerBR));
-        // scale the power between the range of -1 and 1
-        if (scaleFactor > 1) {
-            powerFL /= scaleFactor;
-            powerFR /= scaleFactor;
-            powerBL /= scaleFactor;
-            powerBR /= scaleFactor;
+            motorFR.setZeroPowerBehavior(DcMotor.ZeroPowerBehavior.BRAKE);
+            motorFL.setZeroPowerBehavior(DcMotor.ZeroPowerBehavior.BRAKE);
+            motorBR.setZeroPowerBehavior(DcMotor.ZeroPowerBehavior.BRAKE);
+            motorBL.setZeroPowerBehavior(DcMotor.ZeroPowerBehavior.BRAKE);
+
+            //intakeLeft.setZeroPowerBehavior(DcMotor.ZeroPowerBehavior.BRAKE);
+            //intakeRight.setZeroPowerBehavior(DcMotor.ZeroPowerBehavior.BRAKE);
+
+            //motorLift1.setZeroPowerBehavior(DcMotor.ZeroPowerBehavior.BRAKE);
+            //motorLift2.setZeroPowerBehavior(DcMotor.ZeroPowerBehavior.BRAKE);
+
+            BNO055IMU.Parameters parameters = new BNO055IMU.Parameters();
+            parameters.angleUnit = BNO055IMU.AngleUnit.DEGREES;
+            imu = hardwareMap.get(BNO055IMU.class, "imu");
+            imu.initialize(parameters);
+
+            while (!isStopRequested() && !imu.isGyroCalibrated()){
+                sleep(50);
+                idle();
+            }
         }
 
-        motorFL.setPower(powerFL);
-        motorFR.setPower(powerFR);
-        motorBL.setPower(powerBL);
-        motorBR.setPower(powerBR);
-    }
+        public void driveMecanum(double driveAngle, double turnPower, double drivePower){
+            double x = drivePower * Math.cos(driveAngle);
+            double y = drivePower * Math.sin(driveAngle);
+
+            double powerFL = x + y - turnPower;
+            double powerFR = y - x + turnPower;
+            double powerBL = y - x - turnPower;
+            double powerBR = y + x + turnPower;
+
+            motorFR.setPower(powerFR);
+            motorFL.setPower(powerFL);
+            motorBR.setPower(powerBR);
+            motorBL.setPower(powerBL);
+
+
+        }
+
+        //Used for calculating distance between 2 points
+        public double calculateDistance(double deltaX, double deltaY){
+            return Math.sqrt(Math.pow(deltaX, 2) + Math.pow(deltaY, 2));
+        }
+
+
 
-    //if you subtract 359 from 0, you would get -359 instead of 1
-    //this method handles cases when one angle is multiple rotations from the other
-    public double adjustAngles(double firstAngle, double secondAngle)
-    {
-        double delta = firstAngle - secondAngle;
-        while(delta > 180)
-            delta -= 360;
-        while(delta < -180)
-            delta += 360;
-        return delta;
-    }
 
-    // Used for calculating distances between 2 points
-    double calculateDistance(double deltaX, double deltaY)
-    {
-        return Math.sqrt(Math.pow(deltaX, 2) + Math.pow(deltaY, 2));
     }
 
-}
 
```

Comment:
We have all of teleop finished already so can you just delete any changes you have in teleop?

Reply:
I think you can just delete any changes in masterOpMode because I already figured out the correct drive function and yours is different.